aws/resource_aws_network_interface_sg_attachment_test.go

```diff
@@ -56,7 +56,7 @@ func TestAccAWSNetworkInterfaceSGAttachment_disappears(t *testing.T) {
 				Config: testAccAwsNetworkInterfaceSGAttachmentConfig(rName),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckAWSNetworkInterfaceSGAttachmentExists(resourceName, &networkInterface),
-					testAccCheckAWSENIDisappears(&networkInterface),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsNetworkInterface(), resourceName),
```

Comment:
This test has a new failure:

```
=== CONT  TestAccAWSNetworkInterfaceSGAttachment_disappears
TestAccAWSNetworkInterfaceSGAttachment_disappears: resource_aws_network_interface_sg_attachment_test.go:42: [INFO] Got non-empty plan, as expected
TestAccAWSNetworkInterfaceSGAttachment_disappears: resource_aws_network_interface_sg_attachment_test.go:42: Check 2/2 error: Error deleting ENI: InvalidNetworkInterfaceId.Malformed: Invalid id: "sg-0beb8cdaa5e27fcd2_eni-010a1844f2ddecf3d" (expecting "eni-...")
  status code: 400, request id: e0f2e0d8-d7a8-4096-8dcf-579ac880197c
--- FAIL: TestAccAWSNetworkInterfaceSGAttachment_disappears (87.79s)
```

The resource needs to align with the type:

```suggestion
					testAccCheckResourceDisappears(testAccProvider, resourceAwsNetworkInterfaceSGAttachment(), resourceName),
```

Reply:
--- PASS: TestAccAWSNetworkInterfaceSGAttachment_disappears (186.56s)
Thanks!